File: ts/express/src/repository/board.repo.ts
import POOL from "../util/db.connect";

interface BoardBase {
  title: string;
  content: string;
  tag: string;
}

const PAGE_TAKEN = 15;
const { QUERY, VALUES, SET } = POOL;

export const getB: Function = (idx: number) =>
  QUERY`SELECT * FROM board WHERE idx = ${idx}`;

export const getBList: Function = ({
  page = 1,
  orderBy = "desc",
  sortBy = "createdAt",
}: {
  page: number;
  orderBy: "asc" | "desc";
  sortBy: "createdAt";
}) =>
  QUERY`SELECT * FROM board ORDER BY ${sortBy} ${orderBy} LIMIT ${
    (page - 1) * PAGE_TAKEN
  }, ${PAGE_TAKEN}`;

export const insertB: Function = (data: BoardBase) => QUERY`
  INSERT INTO board ${VALUES(data)}
`;

export const patchB: Function = (idx: number, data: Partial<BoardBase>) =>
  QUERY`UPDATE board ${SET(data)} WHERE idx = ${idx}`;

export const deleteB: Function = (idx: number) =>
  QUERY`DELETE FROM board WHERE idx = ${idx}`;

File: py/flask/repo/board_repo.py
from ..domain import Board


class BoardRepository:
    TAKEN_BY_PAGE = 15

    @staticmethod
    def getB(idx: int):
        return Board.select().where(Board.idx == idx)

    @staticmethod
    def getBList(page: int, direction="asc", sortBy="created_At"):
        return (
            Board.select()
            .limit(BoardRepository.TAKEN_BY_PAGE)
            .offset((page - 1) * BoardRepository.TAKEN_BY_PAGE)
            .orderBy(getattr(getattr(Board, sortBy), direction)())
        )

    @staticmethod
    def createB(**post):
        return Board.create(post)

    @staticmethod
    def patchB(idx: int, **opinion):
        query = Board.update(opinion).where(Board.idx == idx)
        return query.execute()

    @staticmethod
    def deleteB(idx: int):
        to_del = Board.get(Board.idx == idx)
        return to_del.delete_instance()
